daemon/mgr/container_state.go

```diff
@@ -0,0 +1,139 @@
+package mgr
+
+import (
+	"time"
+
+	"github.com/alibaba/pouch/apis/types"
+	"github.com/alibaba/pouch/pkg/utils"
+)
+
+// IsRunning returns container is running or not.
+func (c *Container) IsRunning() bool {
+	c.Lock()
+	defer c.Unlock()
+	return c.State.Status == types.StatusRunning
+}
+
+// IsStopped returns container is stopped or not.
+func (c *Container) IsStopped() bool {
+	c.Lock()
+	defer c.Unlock()
+	return c.State.Status == types.StatusStopped
+}
+
+// IsExited returns container is exited or not.
+func (c *Container) IsExited() bool {
+	c.Lock()
+	defer c.Unlock()
+	return c.State.Status == types.StatusExited
+}
+
+// IsCreated returns container is created or not.
+func (c *Container) IsCreated() bool {
+	c.Lock()
+	defer c.Unlock()
+	return c.State.Status == types.StatusCreated
+}
+
+// IsPaused returns container is paused or not.
+func (c *Container) IsPaused() bool {
+	c.Lock()
+	defer c.Unlock()
+	return c.State.Status == types.StatusPaused
+}
+
+// IsDead returns container is dead or not.
+func (c *Container) IsDead() bool {
+	c.Lock()
+	defer c.Unlock()
+	return c.State.Status == types.StatusDead
+}
+
+// IsRunningOrPaused returns true of container is running or paused.
+func (c *Container) IsRunningOrPaused() bool {
+	c.Lock()
+	defer c.Unlock()
+	if c.State.Status == types.StatusRunning {
+		return true
+	}
+	if c.State.Status == types.StatusPaused {
+		return true
+	}
+	return false
+}
+
+// IsRestarting returns container is restarting or not.
+func (c *Container) IsRestarting() bool {
+	c.Lock()
+	defer c.Unlock()
+	return c.State.Status == types.StatusRestarting
+}
+
+// SetStatusRunning sets a container to be status running.
+// When a container's status turns to StatusStopped, the following fields need updated:
+// Status -> StatusRunning
+// StartAt -> time.Now()
+// Pid -> input param
+// ExitCode -> 0
+func (c *Container) SetStatusRunning(pid int64) {
+	c.Lock()
+	defer c.Unlock()
+	if c.State == nil {
```

Comment:
Maybe we can initialise the container `State`, so we no need to check if it it `nil`, WDYT?

Reply:
Yes, reasonable. Try to use contract programming.